zio-http/src/main/scala/zio/http/model/headers/values/ContentLength.scala

```diff
@@ -0,0 +1,35 @@
+package zio.http.model.headers.values
+
+import scala.util.Try
+
+/**
+ * ContentLength header value
+ */
+sealed trait ContentLength
+
+object ContentLength {
+
+  /**
+   * The Content-Length header indicates the size of the message body, in bytes,
+   * sent to the recipient.
+   */
+  final case class ContentLengthValue(length: Long) extends ContentLength
+
+  /**
+   * The ContentLength header value is invalid.
+   */
+  case object InvalidContentLengthValue extends ContentLength
+
+  def fromContentLength(contentLength: ContentLength): String =
+    contentLength match {
+      case ContentLengthValue(length) => length.toString
+      case InvalidContentLengthValue  => ""
+    }
+
+  def toContentLength(value: String): ContentLength =
+    Try(value.trim.toLong).fold(
```

Comment:
Unfortunately this will involve a tremendous amount of overhead. 

I am fine just accepting this as-is, but we will need to separately optimize all of these decodings.